Document Nest Pub/Sub manual topic creation for new Device Access Console changes (#37130)
home-assistant/home-assistant.github.io: source/_integrations/nest.markdown

You are in control of the information and capabilities exposed to Home Assistant. You can authorize a single device, multiple devices, or different levels of functionality such as motion events, live streams, for any particular device. The integration is flexible enough to adapt based on what you allow.

# Prerequisites

> - The Nest Device Access Console Pub/Sub setup process has changed as of January 23rd 2025. **Please make sure you are using the latest version of Home Assistant.**

Reviewer: ```suggestion
{% important %}
- The Nest Device Access Console Pub/Sub setup process has changed as of January 23rd 2025. **Please make sure you are using the latest version of Home Assistant.**
```

> - The Nest Smart Device Management (SDM) API **requires a US$5 fee**. Before buying, make sure your device is [supported](https://developers.google.com/nest/device-access/supported-devices).

Reviewer: ```suggestion
- The Nest Smart Device Management (SDM) API **requires a US$5 fee**. Before buying, make sure your device is [supported](https://developers.google.com/nest/device-access/supported-devices).
{% endimportant %}
```
oh, sorry. what I was trying to say was if you would consider using the important textbox instead of the caution. it looks something like this.
![image](https://github.com/user-attachments/assets/549f4074-704a-4ef5-be79-f3feaddfbccb)

Author: Yes, I don't have a strong opinion and am fine to just go with your first suggestion of using the standard prerequisites like all other integrations use. (which is now applied)

You are in control of the information and capabilities exposed to Home Assistant. You can authorize a single device, multiple devices, or different levels of functionality such as motion events, live streams, for any particular device. The integration is flexible enough to adapt based on what you allow.

# Prerequisites

> {% caution %}

Reviewer: ```suggestion
```
I think the text boxes could be removed here. The Prerequisites block already indicates that these topics need to be checked or done before anything else.

Author: In theory i agree with you... However, users are typically pretty *furious* over missing this. It has previously been a little subtle and users miss it. There are probably tens of reports on this topic in this repo or in the core repo...

If you still feel this should be changed in the name of consistency happy to make the change but just want to give the context.

Author: Eh, let's go for it and see what happens...

Reviewer: oh, I see. Thank you for the context. Would you be Ok with changing it to important, instead of caution? 
The issue I have with Caution is that for product safety documentation, the signal words Caution and Warning have a predefined/standardized meaning (defined in ANSI Z535.6) I realize this is not safety information per se , but I still don't like the term collision.

![image](https://github.com/user-attachments/assets/65c2bfb3-0dd9-43f0-aea0-a702f9355f5e)

The important text box is also relatively flashy purple/pink, so should be visible.

Author: I don't have any preference on this, will use whatever should be used. I did go ahead and apply your suggestions though. The Prerequisites heading may be clear enough so i'm open to try it.

> The Nest Device Access Console Pub/Sub setup process has changed as of January 23rd 2025. **Please make sure you are using the latest version of Home Assistant.**

Reviewer: ```suggestion
- The Nest Device Access Console Pub/Sub setup process has changed as of January 23rd 2025. **Please make sure you are using the latest version of Home Assistant.**
```

> {% endcaution %}

Reviewer: ```suggestion
```

> {% caution %}

Reviewer: ```suggestion
```

> The Nest Smart Device Management (SDM) API **requires a US$5 fee**. Before buying, make sure your device is [supported](https://developers.google.com/nest/device-access/supported-devices).

Reviewer: ```suggestion
- The Nest Smart Device Management (SDM) API **requires a US$5 fee**. Before buying, make sure your device is [supported](https://developers.google.com/nest/device-access/supported-devices).
```

> {% endcaution %}

Reviewer: ```suggestion
```

You are in control of the information and capabilities exposed to Home Assistant. You can authorize a single device, multiple devices, or different levels of functionality such as motion events, live streams, for any particular device. The integration is flexible enough to adapt based on what you allow.

> {% caution %}

Reviewer: L41-L47: could this be in a Prereq section? Both points seem to be prerequisites.

Author: Sure, done.

Read the warnings on the page before proceeding, including Google Account type limitations.
{% endnote %}

> 2. Click on the button **[Go to the Device Access Console](https://console.nest.google.com/device-access/)**.

Reviewer: ```suggestion
2. Select the button **[Go to the Device Access Console](https://console.nest.google.com/device-access/)**.
```

    ![Screenshot of Device Access Registration](/images/integrations/nest/device_access.png)

> 3. Check the box to "Accept the Terms of Service" and click **Continue to Payment** where you need to pay a fee (currently US$5).

Reviewer: ```suggestion
3. Check the box to "Accept the Terms of Service" and select **Continue to Payment** where you need to pay a fee (currently US$5).
```

    ![Screenshot of accepting terms](/images/integrations/nest/accept_terms.png)

> 4. Now the [Device Access Console](https://console.nest.google.com/device-access/project-list) should be visible. Click on **Create project**.

Reviewer: ```suggestion
4. Now the [Device Access Console](https://console.nest.google.com/device-access/project-list) should be visible. Select  **Create project**.
```

> 5. Give your Device Access project a name and click **Next**.

Reviewer: ```suggestion
5. Give your Device Access project a name and select **Next**.
```

    ![Screenshot of naming a project](/images/integrations/nest/project_name.png)

> 6. Next you will be asked for an *OAuth client ID*  which you created in the previous step and click **Next**.

Reviewer: ```suggestion
6. Next you will be asked for an **OAuth client ID**  which you created in the previous step and select **Next**.
```

    ![Screenshot of Device Access Console OAuth client ID](/images/integrations/nest/device_access_oauth_client_id.png)

> 7. Leave *Enable Events* unchecked for now and **Create project**. You need a Pub/Sub topic

Reviewer: ```suggestion
7. Leave **Enable Events** unchecked for now and **Create project**. You need a Pub/Sub topic
```

  (created in the next section) to enable events. This requires additional setup in the
  Google Cloud Pub/Sub console so we will skip that step for now then come back to it in
  the next section.

8. You now have a *Device Access Project ID* needed by Home Assistant.

{% enddetails %}

{% details "Enable events and Pub/Sub topic [Device Access & Cloud Console]" %}

The Nest Device Access Console Pub/Sub setup process has changed as of January 23rd 2025. **Please make sure you are using the latest version of Home Assistant.**.

This section describes how to configure your Device Access Project with a Pub/Sub topic
to publish events for devices in your home. Home Assistant and the Device Access Project must be configured to use the *Topic Name* otherwise you will not receive events.

If you previously set up events, then your Device Access Project may have already created a topic for you and you can use that topic name. For new projects, or if you disable events, you need to create the topic yourself following the instructions below.

1. Go to the [Pub/Sub Google Cloud Console](https://console.cloud.google.com/cloudpubsub/topic/list).

> 2. Click **Create Topic**.

Reviewer: ```suggestion
2. Select **Create Topic**.
```

> 3. Enter a *Topic ID* such as `home-assistant-nest`. You may leave the default settings.

Reviewer: ```suggestion
3. Enter a **Topic ID** such as `home-assistant-nest`. You may leave the default settings.
```

  ![Screenshot of OAuth confirmation](/images/integrations/nest/cloud_pubsub_create_topic.png)

> 4. Click **Create** to create the topic.

Reviewer: ```suggestion
4. Select **Create** to create the topic.
```

> 5. You now have a *Topic Name* needed by the Device Access Console and Home Assistant. The full *Topic Name* that contains your Cloud Project ID and the *Topic ID* such as `projects/<cloud console id>/topics/home-assistant-nest`.

Reviewer: ```suggestion
5. You now have a **Topic Name** needed by the Device Access Console and Home Assistant. The full **Topic Name** that contains your Cloud Project ID and the **Topic ID** such as `projects/<cloud console id>/topics/home-assistant-nest`.
```

> 6. Next, you need to give the Device Access Console permission to publish to your Topic. From the Pub/Sub Topic page click **Add Principal**.

Reviewer: ```suggestion
6. Next, you need to give the Device Access Console permission to publish to your Topic. From the Pub/Sub Topic page select **Add Principal**.
```

  ![Screenshot of OAuth confirmation](/images/integrations/nest/cloud_pubsub_add_principal.png)

> 7. In *New Principals* enter `sdm-publisher@googlegroups.com`

Reviewer: ```suggestion
7. In **New Principals** enter `sdm-publisher@googlegroups.com`
```

> 8. In *Select a Role* under *Pub/Sub* select *Pub/Sub Publisher* and click *Create*.

Reviewer: ```suggestion
8. In **Select a Role** under **Pub/Sub** select **Pub/Sub Publisher** and **Create**.
```

  ![Screenshot of OAuth confirmation](/images/integrations/nest/cloud_pubsub_add_principal_role.png)

9. Next you can configure the Device Access Console to use this topic. Visit the [Device Access Console](https://console.nest.google.com/device-access/).

> 10. Click the Device Access Project you previously created. It should show the Pub/Sub topic

Reviewer: ```suggestion
10. Select the Device Access Project you previously created. It should show the Pub/Sub topic
```

   as disabled. If there is an existing topic shown, then you may delete it and use
   the one you just created to avoid getting them mixed up.

  ![Screenshot of OAuth confirmation](/images/integrations/nest/device_access_topic_disabled.png)

> 11. Click *...* next to *Pub/Sub topic* then *Enable events with PubSub topic*.

Reviewer: ```suggestion
11. select *...* next to **Pub/Sub topic**, then **Enable events with PubSub topic**.
```

> 12. Enter the full Pub/Sub *Topic Name* and click **Add & Validate**. If you see an error, then

Reviewer: ```suggestion
12. Enter the full Pub/Sub **Topic Name** and select **Add & Validate**. If you see an error, then
```

8. You will now see a page hosted by *My Home Assistant* asking if you would like to *Link account to Home Assistant?* Click **Link Account** to continue.

9. If all went well, you will next configure events and Pub/Sub topic. Nest will attempt
   to automatically find a Pub/sub topic either created by the Device Access Console
   or manually by you.

     ![Screenshot of OAuth confirmation](/images/integrations/nest/config_flow_with_topic.png)

> 10.  If you instead see the error message *No eligible Pub/Sub topics found, please ensure Device Access Console has a Pub/Sub topic.* then follow the steps in the previous section to enable events and create a Pub/Sub topic in another browser tab. Once you have created and configured the topic, you may press on this screen *Submit* to refresh the list of topics and continue.

Reviewer: ```suggestion
10.  If you instead see the error message *No eligible Pub/Sub topics found, please ensure Device Access Console has a Pub/Sub topic.* then follow the steps in the previous section to enable events and create a Pub/Sub topic in another browser tab. Once you have created and configured the topic, you may press on this screen **Submit** to refresh the list of topics and continue.
```
